### PR TITLE
MGMT-15163: Add validation: registry image URI

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
validation example:
```shell
FATAL failed to load asset "Appliance Config": invalid Appliance Config configuration: [imageRegistry.uri: Invalid value: "quay.io/libpod/registry:2.81": Invalid uri: Failed to execute cmd (/usr/bin/podman inspect quay.io/libpod/registry:2.81): Error: error inspecting object: no such object: "quay.io/libpod/registry:2.81"
FATAL : exit status 125, imageRegistry.port: Invalid value: 12345555: registryPort must be between 1024 and 65535]
```

File: pkg/asset/config/appliance_config.go [hunk 1]
```diff
@@ -248,20 +255,33 @@ func (a *ApplianceConfig) validateConfig() field.ErrorList {
...
 	if a.Config.ImageRegistry == nil {
 		return nil
 	}
 
-	// TODO(MGMT-15163): add validation for a.Config.ImageRegistry.URI
+	if a.Config.ImageRegistry.URI != nil {
+		cmd := fmt.Sprintf(PodmanInspect, swag.StringValue(a.Config.ImageRegistry.URI))
+		logrus.Debugf("Running uri validation cmd: %s", cmd)
+		args := strings.Split(cmd, " ")
+		_, err := a.executer.Execute(args[0], args[1:]...)
```
Comment (danielerez) [inline pkg/asset/config/appliance_config.go:268]: nit: we can have it in one line.
i.e. if _, err := a.executer.Execute(args[0], args[1:]...); err != nil...

File: pkg/asset/config/appliance_config.go [hunk 2]
```diff
@@ -248,20 +255,33 @@ func (a *ApplianceConfig) validateConfig() field.ErrorList {
...
+
 	if a.Config.ImageRegistry == nil {
 		return nil
 	}
 
-	// TODO(MGMT-15163): add validation for a.Config.ImageRegistry.URI
+	if a.Config.ImageRegistry.URI != nil {
+		cmd := fmt.Sprintf(PodmanInspect, swag.StringValue(a.Config.ImageRegistry.URI))
+		logrus.Debugf("Running uri validation cmd: %s", cmd)
+		args := strings.Split(cmd, " ")
```
Comment (danielerez) [inline pkg/asset/config/appliance_config.go:267]: not relevant for this PR, but we might want to refactor this into some helper func. or maybe add an 'Execute' interface that gets a cmd and do the split internally:/

Comment (nmagnezi) [reply pkg/asset/config/appliance_config.go:267]: I agree.

File: pkg/asset/config/appliance_config.go (hunk 1 above)
Comment (nmagnezi) [reply pkg/asset/config/appliance_config.go]: done

**Pull request merged**